integrations/fallback_chat/src/haystack_integrations/components/generators/fallback_chat/chat_generator.py

```diff
@@ -0,0 +1,404 @@
+            ttft_task.cancel()
+            try:
+                _ = await ttft_task
+            except Exception:  # noqa: S110
+                # Intentionally ignore TTFT task exceptions - we only care about the main task
+                pass
+            return await task
+
+        if ttft_task in done:
+            # Commit to this generator; await completion with optional cap
+            ttft_task.cancel()
+            try:
+                _ = await ttft_task
+            except Exception:  # noqa: S110
+                # Intentionally ignore TTFT task exceptions - we only care about the main task
+                pass
+            try:
+                return await task
+            except asyncio.TimeoutError as e:
+                # Completion took too long after first token
+                raise e
+
+        # Neither completed nor produced a first token within deadline
+        active["value"] = False
+        for p in pending:
+            p.cancel()
+        try:
+            await task
+        except Exception:  # noqa: S110
+            # Intentionally ignore task exceptions - we're timing out and will raise our own error
+            pass
+        msg = "TTFT deadline exceeded for generator (async)"
+        raise asyncio.TimeoutError(msg)
+
+    # ---------------------- Public API ----------------------
+    @component.output_types(replies=list[ChatMessage], meta=dict[str, Any])
+    def run(
+        self,
+        messages: list[ChatMessage],
+        generation_kwargs: dict[str, Any] | None = None,
+        tools: (list[Tool] | Toolset) | None = None,
+        streaming_callback: StreamingCallbackT | None = None,
+    ) -> dict[str, Any]:
+        start = time.time()
+        failed: list[str] = []
+        last_error: BaseException | None = None
+
+        for idx, gen in enumerate(self.generators):
+            gen_name = gen.__class__.__name__
+            effective_timeout = self._get_effective_timeout(gen)
+            try:
+                result = self._run_single_sync(gen, messages, generation_kwargs, tools, streaming_callback)
+                replies = result.get("replies", [])
+                meta = dict(result.get("meta", {}))
+                meta.update(
+                    {
+                        "successful_generator_index": idx,
+                        "successful_generator_class": gen_name,
+                        "total_attempts": idx + 1,
+                        "failed_generators": failed,
+                        "timeout_used": effective_timeout,
+                        "execution_time": time.time() - start,
+                    }
+                )
+                return {"replies": replies, "meta": meta}
+            except asyncio.TimeoutError as e:
+                logger.warning("Generator %s timed out after %.2fs", gen_name, effective_timeout)
+                failed.append(gen_name)
+                last_error = e
```

Comment:
A general comment. I'm not sure I like how we've implemented our own additional timeout management here. I think we should be asking users to set timeouts on each individual ChatGenerator since that is normally specifiable as a timeout param to the ChatGenerator when making it. I think that is more clear and understandable than creating our own mechanism on top

Reply:
Yes, ok we can do that. I'll drop the timeout on the FallbackChatGenerator and the complexities with what takes precendence: list of generators, outer etc